sphinx/writers/latex.py

```diff
@@ -2163,26 +2147,36 @@ def depart_title_reference(self, node):
         # type: (nodes.Node) -> None
         self.body.append('}')
 
+    def visit_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        longest_label = max((subnode[0].astext() for subnode in node), key=len)
+        self.body.append(u'\n\\begin{sphinxthebibliography}{%s}\n' % longest_label)
+
+    def depart_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        self.body.append(u'\\end{sphinxthebibliography}\n')
+
     def visit_citation(self, node):
         # type: (nodes.Node) -> None
-        # TODO maybe use cite bibitems
-        # bibitem: [citelabel, citetext, docname, citeid]
-        self.bibitems.append(['', '', '', ''])
-        self.context.append(len(self.body))
```

Comment:
The responsibility of `Translator` is mapping doctree to the target media. Gathering nodes or transforming doctree  is not. `Transform` module is much better to such works.
After this PR, `visit_citation` behaves as simple representer of citations (citation entries).